be/src/vec/data_types/data_type_object.cpp

```diff
@@ -113,6 +119,8 @@ char* DataTypeObject::serialize(const IColumn& column, char* buf, int be_exec_ve
         // 2.2 serialize subcolumn
         buf = type->serialize(entry->data.get_finalized_column(), buf, be_exec_version);
     }
+    // serialize num of subcolumns
+    *reinterpret_cast<uint32_t*>(size_pos) = num_of_columns;
```

Comment:
variable size_pos is only used once and can be elimilated.

Reply:
no， it's need to be recorded, and assigned then